clio-util/src/main/scala/org/broadinstitute/clio/util/AuthUtil.scala

```diff
@@ -19,44 +17,66 @@ object AuthUtil extends ModelAutoDerivation {
     "https://www.googleapis.com/auth/userinfo.email"
   )
 
+  /** Get an access token by calling out to gcloud. */
+  private def shellOutAuthToken: Either[Throwable, AccessToken] =
+    Try {
+      new AccessToken(Process("gcloud auth print-access-token").!!.trim, null)
+    }.toEither
+
   /**
     * Get google credentials either from the service account json or
     * from the user's default setup.
     *
     * @param serviceAccountPath Option of path to the service account json
     */
-  def getAccessToken(serviceAccountPath: Option[Path]): Try[AccessToken] = {
-    def shellOutAuthToken: AccessToken =
-      new AccessToken(Process("gcloud auth print-access-token").!!.trim, null)
-
+  def getAccessToken(
+    serviceAccountPath: Option[Path]
+  ): Either[Throwable, AccessToken] = {
     serviceAccountPath
       .map { jsonPath =>
-        Try(loadServiceAccountJson(jsonPath))
-          .map(getCredsFromServiceAccount)
-          .recover { case _ => shellOutAuthToken }
+        loadServiceAccountJson(jsonPath)
+          .flatMap(getCredsFromServiceAccount)
+          .left
+          .flatMap(_ => shellOutAuthToken)
       }
-      .getOrElse(Try(shellOutAuthToken))
+      .getOrElse(shellOutAuthToken)
   }
 
-  def loadServiceAccountJson(serviceAccountPath: Path): ServiceAccount = {
-    if (Files.exists(serviceAccountPath)) {
-      val jsonBlob =
-        Source.fromFile(serviceAccountPath.toFile).mkString.stripMargin
-      decode[ServiceAccount](jsonBlob).fold({ error =>
-        throw new RuntimeException(
-          s"Could not decode service account JSON at $serviceAccountPath",
-          error
+  /** Load JSON for a google service account. */
+  def loadServiceAccountJson(
+    serviceAccountPath: Path
+  ): Either[Throwable, ServiceAccount] = {
+    val accountOrErr = for {
+      jsonPath <- Either.cond(
+        Files.exists(serviceAccountPath),
+        serviceAccountPath,
+        new RuntimeException(
+          s"Could not find service account JSON at $serviceAccountPath"
         )
-      }, identity)
-    } else {
-      throw new RuntimeException(
-        s"Could not find service account JSON at $serviceAccountPath"
       )
+      jsonBytes <- Try(Files.readAllBytes(jsonPath)).toEither
+      account <- decode[ServiceAccount](new String(jsonBytes).stripMargin)
+    } yield {
+      account
     }
-  }
 
-  def getCredsFromServiceAccount(serviceAccount: ServiceAccount): AccessToken = {
-    val creds = serviceAccount.credentialForScopes(authScopes)
-    creds.refreshAccessToken()
+    accountOrErr.left.map { error =>
+      new RuntimeException(
+        s"Could not decode service account JSON at $serviceAccountPath",
+        error
+      )
+    }
   }
+
+  /**
+    * Request an access token for the given service account with the
+    * scopes needed to talk through Clio's OpenIDC proxy.
+    */
+  def getCredsFromServiceAccount(
```

Comment:
Can this be private?

Reply:
Nope, it's used in `EnvIntegrationSpec` since we load the JSON from vault instead of from disk there.